src/content/docs/products/ontrack/documentation/panopoto-integration/panopto-upload-process.md

```diff
@@ -0,0 +1,177 @@
+# Panopto Video Upload Process
+
+## Overview
+This document outlines key information about the process of uploading videos to Panopto using the Panopto REST API. While this isn't a comprehensive guide, it highlights the steps involved in creating a session, performing a multipart upload of the video file, creating the manifest, and finalising the upload. This serves as an informational resource for implementing the video upload functionality in the future. Additional research and context will be needed to fully integrate these steps.
+
+---
+
+## Step 1: Create a Session for Video Upload
+
+Before uploading a video, a session must be created. The session provides the necessary details for the upload process, including the upload target and session ID.
+
+###  Call the Session Creation API
+The session creation process is triggered by sending a POST request to the `sessionUpload` endpoint.
+
+- **API Endpoint:**
+  ```ruby
+  POST https://{server}/Panopto/PublicAPI/REST/sessionUpload
+  ```
+- **Payload:**
+  ```http
+  {
+    "FolderId": "{folder_id}"
+  }
+
+  ```
+- **Example Session Creation:**
+  ```ruby
+  response = RestClient.post(
+    "https://#{server}/Panopto/PublicAPI/REST/sessionUpload",
+    { "FolderId" => folder_id }.to_json,
+    { content_type: :json, accept: :json, authorization: "Bearer #{access_token}" }
+  )
+  ```
+
+- **Example Response:**
+  ```http
+  {
+    "ID": "session_id",
+    "UploadTarget": "upload_target_url",
+    "FolderId": "{folder_id}"
+  }
+  ```
+## Step 2: Upload Video via Multipart Upload
+Panopto supports multipart uploads for large video files. This step uploads the video in parts to the ```upload target``` specified in the session creation response.
+
+#### 1. Initiate Multipart Upload:
+- Get the Upload URL: After creating a session, extract the ```UploadTarget``` URL from the session creation response. This URL specifies where the video should be uploaded.
+
+- Set Up AWS SDK : Even though Panopto doesn’t require AWS credentials, use AWS SDK to handle the multipart upload. Initialise the SDK with dummy credentials (to bypass authentication but still use the multipart upload logic).
```

Comment:
Use of standard library is preferred over introducing new wrapper libraries. Specially, when they are platform specific, like AWS S3, which can introduce breaking changes depending on the direction the platform takes.

See the second answer in this post to see how a multipart file upload can be done with standard library in ruby. https://stackoverflow.com/a/46669328

Reply:
I’ve read the comment and made the changes accordingly. I replaced the use of AWS SDK with Rubys standard library for the multipart file upload, as suggested.